### ISSUE TITLE
Gooseneck flipping/glitching

### ISSUE THREAD

**Issue by glefave0209** [fkk-cz/noire-public#750]
Can we have the ability to flip Goosenecks.?  Something reasonable that would require maybe 4 people to do it

Although the recent improvements have helped a ton in the mines, the goosenecks are still very touchy and the ability to flip them would be a huge relief for the mining community and likely a relief for DOT/PD/assist from the requests

It would also be great for the immersion to see how well it would work in the mines.  I'm confident the majority of us would help out the cause when needed

**Comment by IsaacGucci** [comment]
Anything to help flip them would be really helpful

**Comment by WhiteRhinoU812** [comment]
That would be fantastic!

**Comment by kyzxo** [comment]
Great idea!! it’s a constant pain when one flips and clogs the mines.

**Comment by fkk-cz** [comment]
DOT will soon™ have the ability to flip trailers by attaching chains from the tow truck/flatbed to the trailer. This, combined with invoicing, will help alleviate this issue. Having 'multiple people' flipping a trailer doesn't seem to make much sense, given the weight of these gooseneck trailers:

![image](https://user-images.githubusercontent.com/74131024/165015267-58487a48-7c01-4426-aa53-08c944eb7efa.png)

**Comment by glefave0209** [comment]
Ok, fair enough counter-point on the weight/players flipping, but can we at least request to be able to buy the same equipment as DOT so we could do this ourselves also (in addition to DOT invoicing)

The mining community is a great community and I think this would be great to see everyone pitching in and helping flipped trailers. It's also very realistic for civilians to own this type of equipment IRL.  If the trucks we have can haul a full load of mines with a gooseneck, they should be able to strap in a tow one right-side up.  Even though I rarely have issues anymore, I would happily spend the money on the equipment to help others.  

Time is money in the mines.  Every hour these trailers are flipped over costs us $30,000, so hopefully you can consider some form of us being able to help ourselves as well

**Comment by fkk-cz** [comment]
> Ok, fair enough counter-point on the weight/players flipping, but can we at least request to be able to buy the same equipment as DOT so we could do this ourselves also (in addition to DOT invoicing)

Non-whitelisted civilians are **not** going to have invoicing. This will create tremendous issues with malicious players issuing fraudulent invoices, which will prevent people from doing certain things (buying vehicles, weapons, etc.).

_How_ are people even flipping their trailers in the first place? I can understand the issue where the trailer is improperly connected to the vehicle and dragged misaligned to the vehicle, but other cases seem to be user error, i.e. don't drive so fast when operating a vehicle with a large trailer with multiple attached vehicles.

**Comment by glefave0209** [comment]
My bad, I didn't mean for civilians to invoice.. I just meant that we have the ability to buy the same equipment as DOT (chains and such) so we can flip the trailers ourselves

And yes, you're right that the majority of the time, when trailers glitch it's a user error; however, there's a lot of residual effects from this because it's very tight in the mines/sell mines.  One guy glitches his trailer and then takes out 2-3 bystanders

The easiest fix would be for everyone to be sensible, but there's a lot of innocent bystanders that are doing the right things that are still getting burned.  We just need one consistent avenue to be able to retrieve our trailers when they flip.  Even if we have to pay to claim it at insurance, it's way better than losing out on 30k an hour.

**Comment by fkk-cz** [comment]
> when trailers glitch it's a user error; however, there's a lot of residual effects from this because it's very tight in the mines/sell mines. One guy glitches his trailer and then takes out 2-3 bystanders.. Or somebody trying to reflip their trailer by ramming it takes out somebody else

Which one is it, a bug or user error? The only known issue I've seen is where the trailer is attached incorrectly (not aligned properly) to the vehicle but cannot be detached unless another vehicle is used to detach it. Refer to https://github.com/fkk-cz/noire-public/issues/359.

People driving their vehicles poorly or too fast is not a valid use case for flipping trailers. There is no elegant solution for such, the best case scenario will involve expanding the insurance system or giving DOT the sole ability to move vehicles. Civilians who drive poorly should have to suffer the consequences instead of misusing `/assist` and asking staff to come and delete their flipped vehicles.
